### ISSUE TITLE
UnicodeDecodeError

### ISSUE THREAD

**Issue by wjyzidane** [hyunhwaj/SalmonTE#11]
Hi hyunhwaj,

Previously, SalmonTE works fine for me but recently when I run the program, it gives the error as below and any comments?

~/software/SalmonTE-master/SalmonTE.py quant --reference=m S17_R1_001.fastq.gz S17_R2_001.fastq.gz

![image](https://user-images.githubusercontent.com/34944811/40788647-8deb4dd6-64be-11e8-92fe-019eb453d670.png)

Jingyi

**Comment by wjyzidane** [comment]
Btw, if I use this command, it gives another error:

~/software/SalmonTE-master/SalmonTE.py quant --reference=m S17_R1_001.fastq.gz 

![image](https://user-images.githubusercontent.com/34944811/40788827-f439af38-64be-11e8-86f2-387dc3cbceac.png)

**Comment by hyunhwaj** [comment]
Hello @wjyzidane, 

The issue seems to be related to #7, and I am wondering which version of SalmonTE you are using. If your SalmonTE version is not up-to-date then please update to the latest version. Otherwise, you can modify a part of `SalmonTE.py` as below:

```py
def get_first_readid(file_name):
    with open(file_name, "r", encoding="ISO-8859-1") as inp:
        return inp.readline().split()[0]
```

Best Regards,

Hyun-Hwan Jeong

**Comment by hyunhwaj** [comment]
The second issue, can you check the `EXPR.csv` has been generated and show me the list of files/directories are stored in `SalmonTE_output` directory?

Hyun-Hwan Jeong

**Comment by wjyzidane** [comment]
Btw, if I uncompress the file and run SalmonTE again, it will generate the files as below:

![image](https://user-images.githubusercontent.com/34944811/40800720-263cce50-64df-11e8-99db-cf2d0a0729bf.png)

but still give the error as below:

![image](https://user-images.githubusercontent.com/34944811/40800743-38d76c28-64df-11e8-94de-e21c36218353.png)


Does the file generate completely?

**Comment by hyunhwaj** [comment]
You must see two directories named by `S17_R1_001` and `S17_R2_001` in your `SalmonTE_output` directory if your run has been done without any problems before merging quantification result. Can you see those directories? 

Also, I am suspecting you have an issue with a python package. Can you check you are having any error messages when you run below?

```
python3 # this should be python3
import pandas pd # this will be executed once you are in the command line mode
```
Hyun-Hwan Jeong

**Comment by wjyzidane** [comment]
If I run the pair end unzip file, it gives the error as below, no SalmonTE_output folder generated. I suspect it is because the file name is not correct?

![image](https://user-images.githubusercontent.com/34944811/40804974-dc0a8280-64ea-11e8-9780-9af8a8e1f3b1.png)

If I run the single end unzip file, it gives the error as I reported in the last post. There is SalmonTE folder with a dir called S17_R1_001 generated and files included are shown in the last post.

Looks like the "numpy" script is missing?

![image](https://user-images.githubusercontent.com/34944811/40805089-385c5a90-64eb-11e8-8ced-87a6187e14b0.png)

**Comment by hyunhwaj** [comment]
@wjyzidane Oh, my bad, `import pandas pd` has to be `import pandas as pd`, and I am thinking this problem is because of the missing of `numpy` package, so could you reinstall `numpy` and `pandas`. If you are using `pip` then you can run below command lines for the reinstallation:

```
pip uninstall pandas
pip uninstall numpy
pip install pandas
pip install numpy
```

Hope this helps you.

Hyun-Hwan Jeong

**Comment by wjyzidane** [comment]
Hi Hyun-Hwan,

Thank you so much for your detailed explanation. I figured it out. I have updated my python path because I try to use some package from other people’s folder where the numpy there seems incomplete. I just delete those from my python path and everything is back to normal. Sorry for bothering you. 

Best,
Jingyi

**Comment by wjyzidane** [comment]
Btw, but I still think the name recognition is a problem for pair-end file:


![image](https://user-images.githubusercontent.com/34944811/40850909-db43c43e-6593-11e8-9c27-114d14a6cb2a.png)

**Comment by hyunhwaj** [comment]
Glad to hear that, @wjyzidane !

By the way, I am wondering how you can resolve the first problem. Could you briefly explain what you did to do?

Thank you,

Hyun-Hwan Jeong

**Comment by wjyzidane** [comment]
Actually, I think there are four problems.

1 numpy package is missing. I solved it by clean the path.
2 unicode error, I solved it by modifying the script as you suggested. 
3 the pair filename problem. it is unsolved as I mentioned in the last post 
4 the compressed file problem. There is no error but no output as well as shown below.

![image](https://user-images.githubusercontent.com/34944811/40853129-f38b7a9e-659a-11e8-9085-4849d85bd6e1.png)

![image](https://user-images.githubusercontent.com/34944811/40853143-ffba17da-659a-11e8-8c90-1410126cdc24.png)

![image](https://user-images.githubusercontent.com/34944811/40853160-0ef0598a-659b-11e8-909d-db9343772a0d.png)

I am sorry to report so many problems in one post but I just find them all at the same time.

Jingyi
